File: loop.lua
print( "1. UNCONDITIONAL / UNCONDITIONAL\n" )
for i = 1,4 do 
    print( "i = " .. tostring(i) ) 
    if i > 2 then
        break
    end
end

print( "\n2. LABELED / UNLABELED" )
for i = 1,4 do 
    print( "\ni = " .. tostring(i) ) 
    for j = 1,4 do 
        print( "j = " .. tostring(j) ) 
        if j > 2 then
            break
        end
    end
end

File: loop.rs
fn main() {
    // unconditinal exit
    println!("1. UNCONDITIONAL / UNCONDITIONAL");
    for i in 1..5 {
        println!("i = {i}");
        if i > 2 {
            break;
        }
    }
    
    // unlabeled exit
    println!("\n2. LABELED / UNLABELED");
    for i in 1..5 {
        println!("\ni = {i}");
        for j in 1..5 {
            println!("j = {j}");
            if j > 2 {
                break;
            }
        }
    }
    
    // labeled exits
    println!("\n***");
    'iloop: for i in 1..5 {
        println!("\ni = {i}");
        for j in 1..5 {
            println!("j = {j}");
            if j > 2 {
                break 'iloop;
            }
        }
    }
    
    println!("\n***");
    for i in 1..5 {
        println!("\ni = {i}");
        'jloop: for j in 1..5 {
            println!("j = {j}");
            if j > 2 {
                break 'jloop;
            }
        }
    }
}